sunpy/timeseries/timeseries_factory.py

```diff
@@ -278,6 +278,11 @@ def _sanitise_args(self, args):
                             args.pop(j)
 
                 args[i] = (data, meta, units)
+
+            elif isinstance(arg, str) and is_url(arg):
+                args[i] = Request(path)
```

Comment:
Based on the test failures, I think this should be `arg` and not `path`.

```suggestion
                args[i] = Request(arg)
```

Reply:
Woop, thanks for catching, should be fixed now.